src/components/CustomerQuota/ItemsSelection/ItemsSelectionCard.tsx

```diff
@@ -152,6 +161,13 @@ export const ItemsSelectionCard: FunctionComponent<ItemsSelectionCard> = ({
         setIsVisible={setIsAddUserModalVisible}
         validateAndUpdateIds={onCheckAddedUsers}
       />
+      <PaymentConfirmationModal
+        isVisible={showPaymentAlert}
+        commitCart={commitCart}
+        cancelPayment={() => {
+          cancelCart(() => setShowPaymentAlert(false));
+        }}
+      />
```

Comment:
You could look into replacing this with `showConfirmationAlert`, you could alter it to toggle visibility if you need that functionality.

Reply:
I tried doing this but it was not possible! My attempt is [here in this commit](https://github.com/rationally-app/mobile-application/pull/218/commits/99704b7302b7e752dcf5498eb055a6d56197f121#diff-d40004bc8e92ebfa7cb6f781d83fecc05dc6b4a26e3d06a73107f0971184f750R141)

The problem is when I pass `commitCart` as an argument in the `showConfirmationAlert` function, all the states get captured. Because `showConfirmationAlert()` is called before `reserveCart` is done executing (due to the async nature of js), the state that is captured in `commitCart` is "RESERVING", and we do not get the correct `checkoutResult`.

I tried awaiting `reserveCart` but that didn't work either, because it is wrapped in `useCallback`